db/version_set.cc

```diff
@@ -503,6 +514,12 @@ class FilePickerMultiGet {
     *fd = f;
     *file_index = curr_file_index;
     *is_last_key_in_file = cmp_largest == 0;
+    if (!*is_last_key_in_file) {
+      // If the largest key in the batch overlapping the file is not the
+      // largest key in the file, upper_ley_ would not have been updated so
+      // update it here
+      upper_key_ = batch_iter_;
```

Comment:
can we update `upper_key_` in the above `else` block?

Reply:
It may be slightly more efficient to do it here as we don't need to update every iteration, but just in the end.

Reply:
didn't notice the wrapping while loop, makes sense to me now